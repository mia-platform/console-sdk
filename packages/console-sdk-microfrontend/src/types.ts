/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  Collections,
  ConfigMaps,
  Endpoints,
  IProject,
  PublicVariable,
  Services,
} from '@mia-platform/console-types'
import type { Observable } from 'rxjs'

export enum EventsTypes {
  SET_SELECTED_PROJECT = 'SET_SELECTED_PROJECT',
  HIDE_ENVIRONMENT_SELECTION = 'HIDE_ENVIRONMENT_SELECTION',
  SHOW_ENVIRONMENT_SELECTION = 'SHOW_ENVIRONMENT_SELECTION',
  CHANGE_ENVIRONMENT_SELECTION = 'CHANGE_ENVIRONMENT_SELECTION',
  CHANGE_CONFIG_MAP_FILE = 'CHANGE_CONFIG_MAP_FILE',
  OVERWRITE_CONFIG_MAPS = 'OVERWRITE_CONFIG_MAPS',
  CREATE_SERVICE_ENVIRONMENT_VARIABLES = 'CREATE_SERVICE_ENVIRONMENT_VARIABLES',
  CHANGE_SERVICE_ENVIRONMENT_VARIABLES = 'CHANGE_SERVICE_ENVIRONMENT_VARIABLES',
  DELETE_ENDPOINT = 'DELETE_ENDPOINT',
  UPDATE_PROJECT_LINKS = 'UPDATE_PROJECT_LINKS',
  INVALIDATE_PROJECTS_CACHE = 'INVALIDATE_PROJECTS_CACHE',
  SEND_CONFIGURATION = 'SEND_CONFIGURATION',
  SET_DYNAMIC_ROUTES = 'SET_DYNAMIC_ROUTES'
}

export type Events = {
  action: EventsTypes,
  publisherId: string,
  payload: unknown,
}

export enum ContextsType {
  FEATURE_TOGGLE_CONTEXT = 'featureTogglesProxyContext',
  HOTKEYS_CONTEXT = 'hotkeysContext'
}

export type IContexts = {
  featureTogglesProxyContext: Record<string, unknown>
  hotkeysContext: Record<string, unknown>
}

export type ISDKConsoleObservable = {
  endpoints: Observable<Endpoints>
  collections: Observable<Collections>
  configMaps: Observable<ConfigMaps>
  services: Observable<Services>
  unsecretedVariables: Observable<PublicVariable[]>

  forceConfigUpdateChecksum: Observable<string>
  microfrontendPluginConfig: Observable<Record<string, unknown>>

  selectedEnvironment: Observable<string>
  selectedProject: Observable<IProject>

  _version: string
}

export type IWriteConfig = (payload: unknown) => void

export type ISDKProps = {
  name: string
  container: HTMLElement
  console: {
    writeConfig: IWriteConfig
    _signals: {mount: () => void}
    eventBus: (event: Events) => void
    configObservables: ISDKConsoleObservable
    contexts: IContexts
  }
}

export type IConsoleProps = IContexts & {
  name: string
  container: HTMLElement
  eventListener: (event: Events) => void
  featureTogglesProxyContext: IContexts['featureTogglesProxyContext']
  hotkeysContext: IContexts['hotkeysContext']
  resourceAPI: ISDKConsoleObservable & {
    writeConfig: IWriteConfig
    _signals: {mount: () => void}
  }
}

