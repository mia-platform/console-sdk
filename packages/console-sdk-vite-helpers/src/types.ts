declare module 'vite-plugin-qiankun/dist/helper' {
    interface QiankunProps extends IConsoleProps {}
  }
  