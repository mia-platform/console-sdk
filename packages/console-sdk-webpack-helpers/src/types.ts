import { IConsoleProps } from '@mia-platform/console-sdk-microfrontend'

declare module 'vite-plugin-qiankun/dist/helper' {
  // IConsoleProps are passed from Quiankun across QuiankusProps to microfrontend 
  interface QiankunProps extends IConsoleProps {}
}
  