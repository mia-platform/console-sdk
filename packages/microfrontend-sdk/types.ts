
export type IMountingProps = {
  container: string
  console: {
    eventBus: (event: Events) => void
  }
}

export type IContextType = {
  HOTKEYS: "HOTKEYS"
  FEATURE_TOGGLES: "FEATURE_TOGGLES"
} 

export type IConsoleCommand = {
  SET_SELECTED_PROJECT: "SET_SELECTED_PROJECT",
  HIDE_ENVIRONMENT_SELECTION: "HIDE_ENVIRONMENT_SELECTION",
  SHOW_ENVIRONMENT_SELECTION: "SHOW_ENVIRONMENT_SELECTION",
  GET_CONFIG: "GET_CONFIG", // Deprecated, probably used only from Visualize
  CHANGE_ENVIRONMENT_SELECTION: "CHANGE_ENVIRONMENT_SELECTION",
  CHANGE_CONFIG_MAP_FILE: "CHANGE_CONFIG_MAP_FILE",
  OVERWRITE_CONFIG_MAPS: "OVERWRITE_CONFIG_MAPS",
  CREATE_SERVICE_ENVIRONMENT_VARIABLES: "CREATE_SERVICE_ENVIRONMENT_VARIABLES",
  CHANGE_SERVICE_ENVIRONMENT_VARIABLES: "CHANGE_SERVICE_ENVIRONMENT_VARIABLES",
  DELETE_ENDPOINT: "DELETE_ENDPOINT",
  UPDATE_PROJECT_LINKS: "UPDATE_PROJECT_LINKS",
  INVALIDATE_PROJECTS_CACHE: "INVALIDATE_PROJECTS_CACHE",
  SEND_CONFIGURATION: "SEND_CONFIGURATION"
}

export type IConfigurationScopes = {
  SERVICES: "SERVICES",
  CONFIG_MAPS: "CONFIG_MAPS",
  COLLECTIONS: "COLLECTIONS",
  SELECTED_PROJECT: "SELECTED_PROJECT",
  ENDPOINTS: "ENDPOINTS",
  UNSECRETED_VARIABLES: "UNSECRETED_VARIABLES",
  SELECTED_ENVIRONMENT: "SELECTED_ENVIRONMENT",
  FAST_DATA: "FAST_DATA",
  MICROFRONTEND_PLUGIN: "MICROFRONTEND_PLUGIN",
  FORCE_CONFIG_UPDATE_CHECKSUM: "FORCE_CONFIG_UPDATE_CHECKSUM"
}

enum EventsTypes {
  'SET_DYNAMIC_ROUTES'
}

export interface Events {
  action: EventsTypes,
  publisherId: string,
  payload: unknown,
}
