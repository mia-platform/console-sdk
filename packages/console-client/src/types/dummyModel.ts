export type DummyModel = {
  id: string
}
