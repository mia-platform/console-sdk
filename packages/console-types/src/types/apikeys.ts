/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { FromSchema } from 'json-schema-to-ts'

import { VALIDATION_ERROR_ID } from '../strings'
import { description } from './shared'

export const apiKeyString = {
  type: 'string',
  minLength: 1,
  pattern: '^[^\\s]+$',
  [VALIDATION_ERROR_ID]: 'apiKey.patternError',
} as const

export const clientType = {
  type: 'string',
  pattern: '^[a-zA-Z_][A-Za-z0-9_]*$',
  [VALIDATION_ERROR_ID]: 'clientType.patternError',
} as const

export const apiKey = {
  type: 'object',
  properties: {
    secret: apiKeyString,
    active: { type: 'boolean' },
    description,
    clientType,
  },
  required: ['secret', 'active', 'clientType'],
  additionalProperties: false,
} as const
export type APIKey = FromSchema<typeof apiKey>

// @Deprecated
export const secret = apiKey

export const apiKeys = {
  type: 'array',
  items: apiKey,
  default: [],
} as const

// @Deprecated
export const secrets = apiKeys
