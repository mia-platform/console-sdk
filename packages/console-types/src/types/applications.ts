import { collectionName } from './collections'
import { endpoint } from './endpoints'
import { serviceName } from './services'

const applicationName = {
  type: 'string',
  minLength: 1,
  pattern: '^[a-z]([-a-z0-9]*[a-z0-9])?$',
}

export const application = {
  type: 'object',
  properties: {
    name: { type: 'string' },
    description: { type: 'string' },
    generatedFrom: { type: 'string' },
    resources: {
      type: 'object',
      properties: {
        services: {
          type: 'object',
          patternProperties: {
            [serviceName.pattern]: {
              type: 'object',
              properties: {
                name: { type: 'string' },
              },
              required: ['name'],
              additionalProperties: true,
            },
          },
        },
        endpoints: {
          type: 'object',
          patternProperties: {
            [endpoint.basePath.pattern]: {
              type: 'object',
              properties: {
                basePath: { type: 'string' },
                service: { type: 'string' },
              },
              required: ['basePath', 'service'],
              additionalProperties: true,
            },

          },
        },
        collections: {
          type: 'object',
          patternProperties: {
            [collectionName.pattern]: {
              type: 'object',
              properties: {
                name: { type: 'string' },
              },
              required: ['name'],
              additionalProperties: true,
            },
          },
        },
        unsecretedVariables: {
          type: 'object',
          patternProperties: {
            '.*': {
              type: 'object',
              properties: {
                name: { type: 'string' },
              },
              required: ['name'],
              additionalProperties: true,
            },
          },
        },
      },
    },
  },
} as const

export const applications = {
  type: 'object',
  patternProperties: {
    [applicationName.pattern]: application,
  },
  additionalProperties: false,
  default: {},
} as const
