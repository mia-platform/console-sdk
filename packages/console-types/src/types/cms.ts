/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { FromSchema } from 'json-schema-to-ts'

import { CHART, CHART_SUMMARY, CMS_INTERFACE_TYPES, CMS_LAYOUT_TYPES, CMS_PROPERTY_VISIBILITY_LEVELS, CUSTOM_STOCK } from '../constants/cms'
import { buildType, description } from './shared'
import { collectionFieldName, collectionName } from './collections'
import { aclExpression } from './endpoints'

const cmsName = {
  type: 'string',
  minLength: 1,
  maxLength: 32,
  pattern: '(^[a-zA-Z0-9_\\-\\s]+$)',
} as const

const positiveInteger = {
  type: 'integer',
  minimum: 0,
} as const

const layout = {
  type: 'string',
  enum: CMS_LAYOUT_TYPES.map((layout) => layout.value),
} as const

const cmsVisibilityFields = {
  type: 'object',
  properties: {
    query: {
      type: 'string',
    },
  },
  additionalProperties: false,
} as const

const cmsVisibilityLevel = {
  type: 'integer',
  enum: CMS_PROPERTY_VISIBILITY_LEVELS.map((level) => level.value),
} as const

const cmsVisibility = {
  type: 'object',
  properties: {
    level: cmsVisibilityLevel,
    new: cmsVisibilityFields,
    edit: cmsVisibilityFields,
  },
  additionalProperties: false,
  required: ['level'],
} as const

const interfaceType = {
  type: 'string',
  enum: CMS_INTERFACE_TYPES.map((type) => type.value),
} as const

const textLookup = {
  type: 'object',
  properties: {
    sources: { type: 'array', items: { type: 'string' } },
    delimiters: { type: 'array', items: { type: 'string' } },
  },
  additionalProperties: false,
  required: ['sources', 'delimiters'],
} as const

const cmsLookup = {
  type: 'object',
  properties: {
    text: textLookup,
    value: { type: 'string' },
    endpoint: { type: 'string' },
    inlineSource: {
      type: 'array',
      items: {
        type: 'object',
        properties: {
          label: { type: 'string' },
        },
        additionalProperties: true,
      },
    },
    autoSelect: { type: 'boolean' },
    reset: { type: 'boolean' },
    limit: positiveInteger,
    query: { type: 'string' },
    searchLive: { type: 'boolean' },
  },
  additionalProperties: false,
} as const

const property = {
  id: collectionFieldName,
  label: { type: 'string', minLength: 1 },
  interfaceType,
  cmsDateFormat: { type: 'string' },
  cmsLookup,
  cmsZoom: positiveInteger,
  description,
  cmsOrder: positiveInteger,
  cmsCardPosition: positiveInteger,
  cmsEditable: { type: 'boolean' },
  cmsVisibility,
} as const

export const cmsProperty = {
  type: 'object',
  properties: property,
  additionalProperties: false,
  required: ['id', 'label', 'interfaceType', 'cmsVisibility'],
} as const

const properties = {
  type: 'object',
  patternProperties: {
    [cmsName.pattern]: cmsProperty,
  },
  additionalProperties: false,
} as const

const action = {
  type: 'object',
  properties: {
    id: { type: 'string' },
    label: { type: 'string' },
    route: { type: 'string' },
    icon: { type: 'string' },
  },
} as const

const pageAnalytics = {
  type: 'object',
  required: ['id'],
  properties: {
    id: {
      type: 'string',
    },
    width: {
      type: 'integer',
      enum: [6, 12],
      default: 6,
    },
    order: {
      type: 'integer',
      minimum: 0,
      default: 0,
    },
  },
  additionalProperties: false,
} as const

const cmsPageAnalytics = {
  type: 'array',
  items: pageAnalytics,
  default: [],
} as const

const highlight = {
  type: 'object',
  properties: {
    query: { type: 'string' },
    color: { type: 'string' },
    backgroundColor: { type: 'string' },
  },
  additionalProperties: false,
} as const

const notification = {
  type: 'object',
  properties: {
    query: { type: 'string' },
  },
  additionalProperties: false,
} as const

const page = {
  actions: { type: 'array', items: action },
  id: { type: 'string' },
  name: cmsName,
  endpoint: { type: 'string', minLength: 1 },
  icon: { type: 'string' },
  order: positiveInteger,
  layout,
  highlight,
  notification,
  baseQuery: { type: 'string' },
  defaultDelete: { type: 'boolean' },
  blocked: { type: 'boolean' },
  hidden: { type: 'boolean' },
  properties,
  analytics: cmsPageAnalytics,
  aclExpression: { type: 'string' },
} as const

export const cmsPage = {
  type: 'object',
  properties: page,
  additionalProperties: false,
  required: ['id', 'name', 'endpoint', 'layout', 'properties'],
} as const

const pages = {
  type: 'object',
  patternProperties: {
    [cmsName.pattern]: cmsPage,
  },
  additionalProperties: false,
} as const

export const cmsCategory = {
  id: { type: 'string' },
  name: cmsName,
  order: positiveInteger,
  pages,
} as const

export const cmsCategories = {
  type: 'object',
  patternProperties: {
    [cmsName.pattern]: {
      type: 'object',
      properties: cmsCategory,
      additionalProperties: false,
      required: ['id', 'name', 'order', 'pages'],
    },
  },
  default: {},
  additionalProperties: false,
} as const

export const cmsButton = {
  type: 'object',
  properties: {
    id: { type: 'string', minLength: 1 },
    label: { type: 'string', minLength: 1 },
    route: { type: 'string', minLength: 1, pattern: '^[\\/]\\w*' },
    icon: { type: 'string' },
  },
  required: ['id', 'label', 'route'],
  additionalProperties: false,
} as const

export const cmsSettings = {
  type: 'object',
  required: ['accessGroupsExpression'],
  properties: {
    accessGroupsExpression: {
      type: aclExpression.type,
      default: 'isBackoffice && groups.admin',
    },
  },
  default: {
    accessGroupsExpression: 'isBackoffice && groups.admin',
  },
  additionalProperties: false,
} as const

const yAxis = {
  type: 'object',
  required: ['shared', 'title'],
  properties: {
    shared: { type: 'boolean' },
    labelType: { type: 'string', enum: ['euro', ''] },
    title: { type: 'string' },
  },
  additionalProperties: false,
} as const

const defaultZoom = {
  type: 'integer',
  minimum: 0,
  maximum: 6,
} as const

const series = {
  type: 'object',
  required: ['id', 'name', 'type', 'color'],
  properties: {
    id: { type: 'string' },
    name: { type: 'string' },
    type: { type: 'string', enum: ['line', 'area', 'spline', 'column'] },
    color: { type: 'string' },
    labelType: { type: 'string', enum: ['euro', ''] },
    params: {
      type: 'object',
      required: ['collection'],
      properties: {
        collection: collectionName,
        group: { type: 'string' },
        groupDate: { type: 'string' },
        format: { type: 'string', enum: ['y', 'ym', 'yw', 'ymd', 'ymdh', 'ymdhM'] },
        filter: { type: 'string' },
        operations: {
          type: 'array',
          items: {
            type: 'array',
            items: { type: 'string' },
          },
        },
      },
      additionalProperties: false,
    },
  },
  additionalProperties: false,
} as const

const customStockAnalytics = {
  type: 'object',
  properties: {
    type: { type: 'string', const: CUSTOM_STOCK },
    title: { type: 'string' },
    id: { type: 'string' },
    legend: { type: 'boolean' },
    yAxis,
    customRangeDates: { type: 'boolean' },
    defaultZoom,
    series: {
      type: 'array',
      items: series,
    },
  },
  additionalProperties: false,
  required: ['type', 'title', 'legend', 'series'],
} as const

const chartAnalytics = {
  type: 'object',
  properties: {
    type: { type: 'string', enum: [CHART, CHART_SUMMARY] },
    title: { type: 'string' },
    id: { type: 'string' },
    legend: { type: 'boolean' },
    sortBy: {
      type: 'string',
      enum: ['label-asc', 'label-desc', 'value-asc', 'value-desc'],
    },
    customRangeDates: { type: 'boolean' },
    startDate: { type: 'integer' },
    endDate: { type: 'integer' },
    series: {
      type: 'array',
      items: series,
    },
  },
  additionalProperties: false,
  required: ['type', 'title', 'legend', 'series'],
} as const

export const analytics = {
  [CHART]: chartAnalytics,
  [CHART_SUMMARY]: chartAnalytics,
  [CUSTOM_STOCK]: customStockAnalytics,
} as const

export const cmsAnalytics = {
  type: 'object',
  additionalProperties: {
    type: 'object',
    if: buildType({ type: 'string', const: CUSTOM_STOCK }),
    then: customStockAnalytics,
    else: {
      if: buildType({ type: 'string', enum: [CHART, CHART_SUMMARY] }),
      then: chartAnalytics,
      else: false,
    },
  },
  default: {},
} as const

export type CmsAnalytics = FromSchema<typeof cmsAnalytics, {
  parseIfThenElseKeywords: true
}>

export const cmsDashboard = {
  type: 'array',
  items: pageAnalytics,
  default: [],
} as const
