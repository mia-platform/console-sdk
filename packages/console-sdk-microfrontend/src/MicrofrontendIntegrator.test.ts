/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { JSDOM } from 'jsdom'

import { ContextsType, Events, EventsTypes, IConsoleProps } from './types'
import ConsoleSDK from './MicrofrontendIntegrator'

const { window } = new JSDOM(`<body></body>`)
const htmlElement: HTMLElement = window.document.createElement('div')
htmlElement.id = '__quiankun__'

const eventBus = jest.fn()
const quiankunProps = { name: 'testMicrofrontend', container: htmlElement }
const consoleProps: IConsoleProps = {
  ...quiankunProps,
  eventListener: eventBus,
  featureTogglesProxyContext: { updateLocalCache: jest.fn() },
  hotkeysContext: { anyProp: 'anyValue' },
  resourceAPI: {
    endpoints: {},
    collections: {},
    configMaps: {},
    services: {},
    unsecretedVariables: [],

    forceConfigUpdateChecksum: '',
    microfrontendPluginConfig: {},

    selectedEnvironment: {
      envId: 'testEnvId',
    },
    selectedProject: {
      projectId: 'testProjectId',
      name: 'testProjectName',
      _id: 'projectOid',
      environments: [],
      configurationGitPath: '',
      repository: {},
    },

    writeConfig: jest.fn(),

    _version: '0.0.0',
    _signals: { mount: jest.fn() },
  },
}

describe('ConsoleSDK', () => {
  it('should get the contai', () => {
    const microfrontendIntegrator = new ConsoleSDK(consoleProps)
    expect(microfrontendIntegrator.getMicrofrontendNode()).toMatchSnapshot()
  })

  it('should get the container id', () => {
    const microfrontendIntegrator = new ConsoleSDK(consoleProps)
    expect(microfrontendIntegrator.getContainerId()).toBe('testMicrofrontend')
  })

  it('should send an event through the event bus', () => {
    const microfrontendIntegrator = new ConsoleSDK(consoleProps)
    const event: Events = {
      payload: { test: 'case' },
      publisherId: 'microfrontend-test',
      action: EventsTypes.SET_DYNAMIC_ROUTES,
    }

    microfrontendIntegrator.sendEvent(event)
    expect(consoleProps.eventListener).toHaveBeenCalledWith(event)
  })

  it('should get a context', () => {
    const microfrontendIntegrator = new ConsoleSDK(consoleProps)

    const featureToggleContext = microfrontendIntegrator.getContext(ContextsType.FEATURE_TOGGLE_CONTEXT)
    const hotkeysContext = microfrontendIntegrator.getContext(ContextsType.HOTKEYS_CONTEXT)
    const invalidContext = microfrontendIntegrator.getContext('INVALID_CONTEXT' as ContextsType)

    expect(featureToggleContext).toEqual({ updateLocalCache: expect.any(Function) })
    expect(hotkeysContext).toEqual({ anyProp: 'anyValue' })
    expect(invalidContext).toBeUndefined()
  })

  it('should get the console config observable', () => {
    const microfrontendIntegrator = new ConsoleSDK(consoleProps)
    const configObservable = microfrontendIntegrator.getConsoleConfigObservable()

    expect(configObservable).toHaveProperty('endpoints')
    expect(configObservable).toHaveProperty('collections')
    expect(configObservable).toHaveProperty('configMaps')
    expect(configObservable).toHaveProperty('services')
    expect(configObservable).toHaveProperty('unsecretedVariables')
    expect(configObservable).toHaveProperty('selectedProject')
    expect(configObservable).toHaveProperty('selectedEnvironment')
    expect(configObservable).toHaveProperty('_version')
    expect(configObservable).toHaveProperty('forceConfigUpdateChecksum')
    expect(configObservable).toHaveProperty('microfrontendPluginConfig')
  })
})
