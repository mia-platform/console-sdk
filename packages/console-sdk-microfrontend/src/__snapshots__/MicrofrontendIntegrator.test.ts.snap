// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConsoleSDK should get the contai 1`] = `
<div
  id="__quiankun__"
/>
`;
